x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.ts

```diff
@@ -0,0 +1,156 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors, setErrorMessage } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { EngineLogic } from '../engine';
+
+import { POLLING_DURATION, POLLING_ERROR_MESSAGE } from './constants';
+import { ApiLogsData, ApiLog } from './types';
+import { getDateString } from './utils';
+
+interface ApiLogsValues {
+  dataLoading: boolean;
+  apiLogs: ApiLog[];
+  meta: ApiLogsData['meta'];
+  hasNewData: boolean;
+  polledData: ApiLogsData;
+  intervalId: number | null;
+}
+
+interface ApiLogsActions {
+  fetchApiLogs(options?: { isPoll: boolean }): { isPoll: boolean };
+  pollForApiLogs(): void;
+  onPollStart(intervalId: number): { intervalId: number };
+  onPollInterval(data: ApiLogsData): ApiLogsData;
+  storePolledData(data: ApiLogsData): ApiLogsData;
+  updateView(data: ApiLogsData): ApiLogsData;
+  onUserRefresh(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+}
+
+export const ApiLogsLogic = kea<MakeLogicType<ApiLogsValues, ApiLogsActions>>({
+  path: ['enterprise_search', 'app_search', 'api_logs_logic'],
+  actions: () => ({
+    fetchApiLogs: ({ isPoll } = { isPoll: false }) => ({ isPoll }),
+    pollForApiLogs: true,
+    onPollStart: (intervalId) => ({ intervalId }),
+    onPollInterval: ({ results, meta }) => ({ results, meta }),
+    storePolledData: ({ results, meta }) => ({ results, meta }),
+    updateView: ({ results, meta }) => ({ results, meta }),
+    onUserRefresh: true,
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        updateView: () => false,
+        onPaginate: () => true,
+      },
+    ],
+    apiLogs: [
+      [],
+      {
+        updateView: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
+      {
+        updateView: (_, { meta }) => meta,
+        onPaginate: (state, { newPageIndex }) => updateMetaPageIndex(state, newPageIndex),
+      },
+    ],
+    hasNewData: [
+      false,
+      {
+        storePolledData: () => true,
+        updateView: () => false,
+      },
+    ],
+    polledData: [
+      {} as ApiLogsData,
+      {
+        storePolledData: (_, data) => data,
+      },
+    ],
+    intervalId: [
+      null,
+      {
+        onPollStart: (_, { intervalId }) => intervalId,
+      },
+    ],
+  }),
+  listeners: ({ actions, values }) => ({
+    pollForApiLogs: () => {
+      if (values.intervalId) return; // Ensure we only have one poll at a time
+
+      const id = window.setInterval(() => actions.fetchApiLogs({ isPoll: true }), POLLING_DURATION);
+      actions.onPollStart(id);
+    },
```

Comment:
Changed from the standalone UI:

- This now uses `setInterval` instead of `setTimeout` to poll, which actually fixes an issue in our current UI where XHRs are getting duplicated (`api_logs` is called twice every 5s instead of just once 😬)
- I separated the poll start into its own action, which I think will be clearer why in a little bit in the actual view/useEffect usage